src/sgame/components/MedipadComponent.cpp

```diff
@@ -1,13 +1,179 @@
+/*
+===========================================================================
+
+Unvanquished GPL Source Code
+Copyright (C) 1999-2005 Id Software, Inc.
+Copyright (C) 2000-2009 Darklegion Development
+
+This file is part of the Unvanquished GPL Source Code (Unvanquished Source Code).
+
+Unvanquished is free software; you can redistribute it
+and/or modify it under the terms of the GNU General Public License as
+published by the Free Software Foundation; either version 2 of the License,
+or (at your option) any later version.
+
+Unvanquished is distributed in the hope that it will be
+useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Unvanquished; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+===========================================================================
+*/
 #include "MedipadComponent.h"
+#include "sgame/Entities.h"
 
 MedipadComponent::MedipadComponent(Entity& entity, HumanBuildableComponent& r_HumanBuildableComponent)
 	: MedipadComponentBase(entity, r_HumanBuildableComponent)
-{}
+{
+	REGISTER_THINKER(Think, ThinkingComponent::SCHEDULER_AVERAGE, 100);
+}
 
 void MedipadComponent::HandleDie(gentity_t* /*killer*/, meansOfDeath_t /*meansOfDeath*/) {
 	// Clear target's healing flag.
 	// TODO: This will fail if multiple sources try to control the flag.
-	if (entity.oldEnt->target && entity.oldEnt->target->client) {
-		entity.oldEnt->target->client->ps.stats[STAT_STATE] &= ~SS_HEALING_2X;
+	if (target_) {
+		target_->client->ps.stats[STAT_STATE] &= ~SS_HEALING_2X;
+	}
+}
+
+gentity_t* MedipadComponent::GetTarget() const
+{
+	return target_.get();
```

Comment:
Should this return `nullptr` if `target_` is `nullptr`?

Reply:
Yes it should, and does.